lapce-core/src/language.rs

```diff
@@ -4,179 +4,388 @@ use tree_sitter::{Parser, TreeCursor};
+    },
+    #[cfg(feature = "lang-java")]
+    Setting{
+        id: LapceLanguage::Java,
+        language: tree_sitter_java::language,
+        highlight: tree_sitter_java::HIGHLIGHT_QUERY,
+        comment: "//",
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+    },
+];
+
 impl LapceLanguage {
     pub fn from_path(path: &Path) -> Option<LapceLanguage> {
         let extension = path.extension()?.to_str()?.to_lowercase();
         Some(match extension.as_str() {
+            #[cfg(feature = "lang-rust")]
             "rs" => LapceLanguage::Rust,
+            #[cfg(feature = "lang-javascript")]
             "js" => LapceLanguage::Javascript,
+            #[cfg(feature = "lang-javascript")]
             "jsx" => LapceLanguage::Jsx,
+            #[cfg(feature = "lang-typescript")]
             "ts" => LapceLanguage::Typescript,
+            #[cfg(feature = "lang-typescript")]
             "tsx" => LapceLanguage::Tsx,
+            #[cfg(feature = "lang-go")]
             "go" => LapceLanguage::Go,
+            #[cfg(feature = "lang-python")]
             "py" => LapceLanguage::Python,
+            #[cfg(feature = "lang-toml")]
             "toml" => LapceLanguage::Toml,
+            #[cfg(feature = "lang-php")]
             "php" => LapceLanguage::Php,
+            #[cfg(feature = "lang-elixir")]
             "ex" | "exs" => LapceLanguage::Elixir,
+            #[cfg(feature = "lang-c")]
             "c" | "h" => LapceLanguage::C,
+            #[cfg(feature = "lang-cpp")]
             "cpp" | "cxx" | "cc" | "c++" | "hpp" | "hxx" | "hh" | "h++" => {
                 LapceLanguage::Cpp
             }
+            #[cfg(feature = "lang-json")]
             "json" => LapceLanguage::Json,
+            #[cfg(feature = "lang-md")]
             "md" => LapceLanguage::Markdown,
+            #[cfg(feature = "lang-ruby")]
             "rb" => LapceLanguage::Ruby,
+            #[cfg(feature = "lang-html")]
             "html" | "htm" => LapceLanguage::Html,
+            #[cfg(feature = "lang-java")]
             "java" => LapceLanguage::Java,
             _ => return None,
         })
     }
 
-    pub fn comment_token(&self) -> &str {
-        match self {
-            LapceLanguage::Rust => "//",
-            LapceLanguage::Go => "//",
-            LapceLanguage::Javascript => "//",
-            LapceLanguage::Jsx => "//",
-            LapceLanguage::Typescript => "//",
-            LapceLanguage::Tsx => "//",
-            LapceLanguage::Python => "#",
-            LapceLanguage::Toml => "#",
-            LapceLanguage::Php => "//",
-            LapceLanguage::Elixir => "#",
-            LapceLanguage::C => "//",
-            LapceLanguage::Cpp => "//",
-            LapceLanguage::Json => "",
-            LapceLanguage::Markdown => "",
-            LapceLanguage::Ruby => "#",
-            LapceLanguage::Html => "",
-            LapceLanguage::Java => "//",
-        }
+    fn find(&self) -> &Setting {
```

Comment:
`find` is an iterator method. I would like this to be called something else to avoid confusion. Maybe `details` would work?

Reply:
> `find` is an iterator method. I would like this to be called something else to avoid confusion. Maybe `details` would work?

It was actually named as `find_setting` before. I will rename it as `find_properties` or `find_details` depending on the struct name.

Reply:
I don't think you need to "find" something if you know where it is ;) But sure, that's a good solution.